p2p/simulations/adapters/inproc.go

```diff
@@ -177,9 +177,9 @@ func (sn *SimNode) Addr() []byte {
 	return []byte(sn.Node().String())
 }
 
-// Node returns a discover.Node representing the SimNode
-func (sn *SimNode) Node() *discover.Node {
-	return discover.NewNode(sn.ID, net.IP{127, 0, 0, 1}, 30303, 30303)
+// Node returns a node descriptor representing the SimNode
+func (sn *SimNode) Node() *enode.Node {
+	return enode.NewV4(&sn.config.PrivateKey.PublicKey, net.IP{127, 0, 0, 1}, 30303, 30303)
```

Comment:
Hi @fjl, in order for `network.TestHiveStatePersistance` to pass constantly, the Node here has to have the same parameters for both Hive `Register` and `On` calls. `Node()` method should contain only this:

```
return sn.config.Node()
```

This will preserve the port numbers as well.

This looks like an old bug that is now more frequent, for some reason.